### PR TITLE
GTMSessionUploadFetcher: respect GTMSESSION_RECONNECT_BACKGROUND_SESSIONS_ON_LAUNCH flag

### PR DESCRIPTION
**Pull request by maksymmalyhin**
Fix https://github.com/google/gtm-session-fetcher/issues/218.

- replicate approach used in [GTMSessionFetcher](https://github.com/google/gtm-session-fetcher/blob/91ed3d188eb95705fef3c249453b81f32dc557d1/Source/GTMSessionFetcher.m#L249) to reduce app startup performance footprint.

@mwyman I didn't really do a lot of testing, so additional verification from your end is appreciated.

### CODE REVIEW

**Comment by maksymmalyhin** [comment]
@mwyman I also have a question as for the default value of the flag `GTMSESSION_RECONNECT_BACKGROUND_SESSIONS_ON_LAUNCH ` defined [here](https://github.com/google/gtm-session-fetcher/blob/91ed3d188eb95705fef3c249453b81f32dc557d1/Source/GTMSessionFetcher.h#L437). What is the reason for the flag being set to `1` only for iOS 13+ devices? From the first glance it looks like it should be preferable behaviour for all iOS, tvOS and macOS devices.

**Comment by mwyman** [comment]
> @mwyman I also have a question as for the default value of the flag `GTMSESSION_RECONNECT_BACKGROUND_SESSIONS_ON_LAUNCH ` defined [here](https://github.com/google/gtm-session-fetcher/blob/91ed3d188eb95705fef3c249453b81f32dc557d1/Source/GTMSessionFetcher.h#L437). What is the reason for the flag being set to `1` only for iOS 13+ devices? From the first glance it looks like it should be preferable behaviour for all iOS, tvOS, and macOS devices and probably watchOS as well (though may require additional code changes to support watchOS case).

Since this was a change in behavior, we gated it on using the latest target OS at the time to avoid an unexpected change in launch behavior; was considered (right or wrong) that projects that had targeted the latest OS at that time were staying more on-top of things.

File: GTMSessionFetcher.podspec [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # to import GTMSessionFetcher via the CocoaPods dependency Manager.
 Pod::Spec.new do |s|
   s.name        = 'GTMSessionFetcher'
-  s.version     = '1.5.0'
+  s.version     = '1.6.0'
```
Comment (mwyman) [inline GTMSessionFetcher.podspec:5]: Please don't bump the version number in this PR. We like to keep that separate from individual code changes.

**Comment by maksymmalyhin** [comment]
> Since this was a change in behavior, we gated it on using the latest target OS at the time to avoid an unexpected change in launch behavior; was considered (right or wrong) that projects that had targeted the latest OS at that time were staying more on-top of things.

I see, that makes sense. 

We would prefer to have it as a default for all Firebase users but it's a bit difficult (not sure if possible at all) to set up without an extra step for developers. I guess, the default behaviour may be updated in conjunction with the major the version bump which will indicate the braking change for the library users. I assume most of the users lock the dependency on a particular major version. WDYT?

**Review by mwyman — CHANGES_REQUESTED** [review]

**Review by mwyman — APPROVED** [review]

**Pull request merged**